apollo-gradle-plugin-incubating/src/main/kotlin/com/apollographql/apollo/gradle/CompilationUnit.kt

```diff
@@ -18,40 +19,8 @@ class CompilationUnit(
     project: Project
 ) {
   val name = "${variantName}${serviceName}"
-  val outputDir: DirectoryProperty = project.objects.directoryProperty()
-  val transformedQueriesDir: DirectoryProperty = project.objects.directoryProperty()
-
-  /**
-   * Sets the output properties for [codegenTask] and this CompilationUnit such that [Provider]s
-   * accessed via this CompilationUnit carry dependencies on [codegenTask]'s outputs.
-   *
-   * @see org.gradle.api.provider.Property.set
-   */
-  internal fun setupOutputProperties(
-      codegenTask: ApolloCodegenTask,
-      willGenerateTransformedQueries: Boolean,
-      project: Project
-  ) {
-    codegenTask.outputDir.apply {
-      set(project.buildDir.child("generated", "source", "apollo", "classes", variantName, serviceName))
-      finalizeValue()
-    }
-    outputDir.apply {
-      set(codegenTask.outputDir)
-      disallowChanges()
-    }
-
-    codegenTask.transformedQueriesOutputDir.apply {
-      if (willGenerateTransformedQueries) {
-        set(project.buildDir.child("generated", "transformedQueries", "apollo", variantName, serviceName))
-      }
-      finalizeValue()
-    }
-    transformedQueriesDir.apply {
-      set(codegenTask.transformedQueriesOutputDir)
-      disallowChanges()
-    }
-  }
+  lateinit var outputDir: Provider<Directory>
+  lateinit var transformedQueriesDir: Provider<Directory>
```

Comment:
I think these being `var`s makes it so that users can set them (which won't actually work, it'll just break things). Maybe take these as constructor arguments, and pass them as `DirectoryProperty`s from ApolloCodegenTask?

Reply:
Yes, definitely better to have that immutable. I guess `Provider<Directory>` is the type we want. I'll change this.

Reply:
Modification pushed. I started moving the "public" API to interfaces in the "api" package and implementations with the 'Default' prefix.